instrumentation/http/src/main/java/brave/http/HttpResponse.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2013-2020 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package brave.http;
+
+import brave.Clock;
+import brave.Response;
+import brave.Span;
+import brave.internal.Nullable;
+import brave.propagation.TraceContext;
+
+/**
+ * Abstract response type used for parsing and sampling of http clients and servers.
+ *
+ * @see HttpClientResponse
+ * @see HttpServerResponse
+ * @since 5.10
+ */
+public abstract class HttpResponse extends Response {
+  /**
+   * Like {@link HttpRequest#method()} except used in response parsing.
+   *
+   * <p>Notably, this is used to create a route-based span name.
+   *
+   * @since 5.10
+   */
+  @Nullable public String method() {
```

Comment:
Just to confirm, we're adding this but may remove it later if we find it works well to pass the request into the response parser?

Reply:
good point.. I'll put a TODO!